grpcbp/client_middlewares.go

```diff
@@ -101,3 +106,191 @@ func ForwardEdgeContextStreaming(ecImpl ecinterface.Interface) grpc.StreamClient
 		return nil, errors.New("grpc.ForwardEdgeContextStreaming: not implemented")
 	}
 }
+
+// PrometheusUnaryClientInterceptor is a client-side interceptor that provides Prometheus
+// monitoring for Unary RPCs.
+//
+// It emits the following metrics:
+//
+// * grpc_client_active_requests gauge with labels:
+//
+//   - grpc_service: the local service slug, serviceSlug arg
+//   - grpc_method: the name of the method called on the gRPC service
+//   - grpc_slug: the name of the remote server the client connects to
+//
+// * grpc_client_latency_seconds histogram and grpc_client_requests_total
+//   counter with labels:
+//
+//   - grpc_service, grpc_method, grpc_slug
+//   - grpc_success: "true" if status is OK, "false" otherwise
+//   - grpc_type: type of request, i.e unary
+//   - grpc_code: the human-readable status code, e.g. OK, Internal, etc
+func PrometheusUnaryClientInterceptor(serviceSlug, serverSlug string) grpc.UnaryClientInterceptor {
+	return func(
+		ctx context.Context,
+		method string,
+		req interface{},
+		reply interface{},
```

Comment:
🔕 the opposite of req is resp usually

Reply:
looks like in grpc they say `reply`, ref: https://pkg.go.dev/google.golang.org/grpc#UnaryClientInterceptor